### PR TITLE
Article harmonisation jc

### PR DESCRIPTION
**Pull request by jinchad**
## enums.py
- removal of white spaces

## models.py
- removed print statements indicating the progress of the researcher node in `generate_keypoints()` for both `Azure` and `HuggingFace`
- modified `generate_key()` to only have article content as a string input
- modified string concatenation in `compiled_points()`

## prompts.py
- heavily modified instructions and examples in researcher prompt under `Azure` to account for new HTML labeling for the article headers
- modified compiler prompt to improve it's keypoints compilation capabilities (WIP)
- added an incomplete general prompt template which will be further improved in future developments

## prompts_archive.py
- added new class `Azure_archived_prompts`, serving as a storage for old Azure prompts
- added method `return_researcher_prompt_ver` to return desired Azure prompt version in the archive
- added previous Azure researcher prompt template before modification

## checks.py
- removed unused import packages from checks.py 

## harmonisation.py
- removed `article_research_counter` from `OriginalArticle` as it is no longer required
- added `ChecksState` into 'RewritingState' under the key `article_evaluation`
- modified `researcher_node` for the new input article structure
- modified `check_all_articles` to no longer include `researcher_node` in the path map
- added some comments under `__main__`
- modified `__main__` to fit new structure

## main.py
- new python file to run both evaluations and harmonisation workflow
- added `start_article_evaluation` to start article evaluation, taking `article title` in a list as the input
- added `start_article_harmonisation` to start article harmonisation, taking ChecksState output from the evaluation step as its input
- added '__main__' to run the complete pipeline from evaluation to harmonisation

## formatters.py
- added `article_list_indexes`, which returns a list of the input article's indexes.
- added `update_header_dict`, which updates the header dictionary in `concat_headers_to_content`
- modified `concat_headers_to_content` to label the headers with their respective HTML tags and include a parent header under article headers, if applicable.

### CODE REVIEW

File: article-harmonisation/agents/models.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
+from agents.enums import MODELS, ROLES
 from config import settings
 from langchain_core.output_parsers import StrOutputParser
 from langchain_core.prompts import ChatPromptTemplate, PromptTemplate
 from langchain_huggingface import HuggingFaceEndpoint
 from langchain_openai import AzureChatOpenAI
 
-from .enums import MODELS, ROLES
 from .prompts import prompt_tool
```
Comment (issacj-17) [inline article-harmonisation/agents/models.py:11]: We will have to standardise this. Let us no longer use relative paths then.

File: article-harmonisation/utils/formatters.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
 CONTENT_BODY = "extracted_content_body"
 EXTRACTED_HEADERS = "extracted_headers"
 ARTICLE_TITLE = "title"
+META_DESC = "category_description"
+
 KEY_PARQUET_INFO = [
     ARTICLE_TITLE,
     "article_category_names",
```
Comment (issacj-17) [inline article-harmonisation/utils/formatters.py:17]: To standardise this.

File: article-harmonisation/agents/models.py (hunk 1 above)
Comment (jinchad) [reply article-harmonisation/agents/models.py:11]: changed back to .enums

File: article-harmonisation/utils/formatters.py (hunk 2 above)
Comment (jinchad) [reply article-harmonisation/utils/formatters.py:17]: I've standardised it by removing KEY_PARQUET_INFO and only keeping the others above.

**Pull request merged**